guides/typography.md

```diff
@@ -0,0 +1,120 @@
+arranged into typography levels. Each level has a `font-size`, `line-height` and `font-weight`. The
+available levels are:
+
+* `display-4`, `display-3`, `display-2` and `display-1` - Used for large, one-off headers, usually
+at the top of the page (e.g. a hero header).
+* `headline` - Corresponds `<h1>` tag.
+* `title` - Corresponds `<h2>` tag.
+* `subheading-2` - Corresponds `<h3>` tag.
+* `subheading-1` - Corresponds `<h4>` tag.
+* `body-2` - Used for bolder body text.
+* `body-1` - Used as the base body text.
+* `caption` - Primarily for smaller body and hint text.
+* `button` - Used for styling buttons.
+
+The typography levels are collected into a typography config which is used to generate the CSS.
+
+[1]: https://material.io/guidelines/style/typography.html
+
+### Usage
+To use the Angular Material typography, you have to include the `Roboto` font with the 300, 400 and
+500 weights. You can host it yourself or include it from Google Fonts:
+
+```html
+<link href="https://fonts.googleapis.com/css?family=Roboto:300,400,500" rel="stylesheet">
+```
+
+Afterwards you have to add the appropriate CSS classes to the elements that you want to style:
+
+```html
+<h1 class="mat-display-1">Jackdaws love my big sphinx of quartz.</h1>
+<h2 class="mat-h2">The quick brown fox jumps over the lazy dog.</h2>
+```
+
+By default, Angular Material doesn't apply any global CSS, however if you want to use the typography
+styles everywhere, you can apply the `mat-typography` class to an element or the `<body>` itself.
+
+```html
+<h1>This header is unstyled</h1>
+
+<section class="mat-typography">
+  <h1>This header will be styled</h1>
+  <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nihil, molestias.</p>
+</section>
+```
+
+### Customization
+If you want to override one or more of the typography levels, or to specify a different font, you
+can use the `mat-typography-level` and `mat-typography-config` functions to create your custom
+config. `mat-typography-level` takes in the desired `font-size`, `line-height` and `font-weight`,
+whereas `mat-typography-config` collects the levels into a configuration object.
+
+Afterwards you can pass the config to the typography mixins. Material uses a few mixins to generate
+the typography styles:
+
+* `angular-material-typography` - Includes the typography styles for all components, as well as the
+base typography.
+* `mat-base-typography` - Includes the typography for body text, headers etc.
+* `mat-{{component}}-typography` - Every Material component has a corresponding typography mixin,
+which allows for its styles to be overwritten (e.g. `mat-checkbox-typography`).
+
+For more details about the typography functions and default config, see the
+[source](https://github.com/angular/material2/blob/master/src/lib/core/typography/_typography.scss).
+
+
+```scss
+// Import the requied mixins.
+@import '~@angular/material/theming';

```

Comment:
@jelbourn this includes all the necessary typography mixins, but I'm not sure whether the file name won't cause some confusion. We could potentially copy it over and call it `typography.scss`.